source/includes/troubleshooting/server-selection-timeout.rst

```diff
@@ -0,0 +1,66 @@
+Driver Throws a Timeout During Server Selection
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Each driver operation requires that you choose a server that
+satisfies the :manual:`server selection criteria
+</core/read-preference-mechanics>`. If you do not select an appropriate
+server within the `server selection timeout <{+new-api-root+}/MongoDB.Driver.Legacy/MongoDB.Driver.MongoServerSettings.ServerSelectionTimeout.html>`__, the driver throws a
```

Comment:
Legacy driver is deprecated and not supported anymore, please reference to the [current driver instead](https://mongodb.github.io/mongo-csharp-driver/3.3.0/api/MongoDB.Driver/MongoDB.Driver.MongoClientSettings.ServerSelectionTimeout.html)